crates/router/src/types/storage/payment_link.rs

```diff
@@ -1 +1,66 @@
-pub use diesel_models::payment_link::{PaymentLink, PaymentLinkNew};
+use async_bb8_diesel::AsyncRunQueryDsl;
+use diesel::{associations::HasTable, ExpressionMethods, QueryDsl};
+pub use diesel_models::{
+    payment_link::{PaymentLink, PaymentLinkNew},
+    schema::payment_link::dsl,
+};
+use error_stack::{IntoReport, ResultExt};
+
+use crate::{
+    connection::PgPooledConn,
+    core::errors::{self, CustomResult},
+    logger,
+};
+#[async_trait::async_trait]
+
+pub trait PaymentLinkDbExt: Sized {
+    async fn filter_by_constraints(
+        conn: &PgPooledConn,
+        merchant_id: &str,
+        payment_link_list_constraints: api_models::payments::PaymentLinkListConstraints,
+    ) -> CustomResult<Vec<Self>, errors::DatabaseError>;
+}
+
+#[async_trait::async_trait]
+impl PaymentLinkDbExt for PaymentLink {
+    async fn filter_by_constraints(
+        conn: &PgPooledConn,
+        merchant_id: &str,
+        payment_link_list_constraints: api_models::payments::PaymentLinkListConstraints,
+    ) -> CustomResult<Vec<Self>, errors::DatabaseError> {
+        let mut filter = <Self as HasTable>::table()
```

Comment:
Where are we handling other contraints like customer_id, starting_after etc.?

Reply:
customer_id is not required for payment_link, and starting after is being handled by `:created_at.eq(created_time));` will remove the duplicate fields and not required fields in paymentLinkConstraints